src/taxes/income-tax.ts

```diff
@@ -498,12 +499,27 @@ function getSurtaxRates(code: ProvinceCode | FederalCode): Rate[] {
     return TAX_BRACKETS[code].SURTAX_RATES;
 }
 
+export function getFederalTaxRates(yearsToInflate: number): Rate[] {
+    const shouldUse2025Rate = yearsToInflate === 0 && now().getFullYear() === 2025;
+    if (shouldUse2025Rate) {
+        return getTaxRates(FEDERAL_CODE).map(
+            (rate, index) => ({ ...rate, RATE: index === 0 ? CA_LOWEST_TAX_RATE_2025 : rate.RATE }),
+        );
+    }
+    return getTaxRates(FEDERAL_CODE);
```

Comment:
Oui c'est redondant de faire de la dé-construction, mais j'ai essayé avec ceci et les tests jest échouaient toujours en disant que le rate était mauvais. Comme si le fait de modifier le rate[0].RATE modifiait l'objet même pour les appels subséquents à la fonction, je n'y comprenais rien pour vrai.
```suggestion
    const shouldUse2025Rate = yearsToInflate === 0 && now().getFullYear() === 2025;
    const rates = getTaxRates(FEDERAL_CODE);
    if (shouldUse2025Rate) {
        rates[0].RATE = CA_LOWEST_TAX_RATE_2025;
    }
    return rates;
```

Reply:
C'est normal, ta fonction `getTaxRates` fini par retourner les références aux objets dans `TAX_BRACKETS`. En fait ça pourrait vraiment mener à des comportements bizarres. Je pense qu'il y aurait (au moins) deux options :
- Rendre `TAX_BRACKETS` (et autres objets du genre) immuable, récursivement
- Chaque accès à `TAX_BRACKETS` qui retourne un objet devrait retourner une copie

Faudrait vérifier si vous avez d'autres objets de constants comme ça qui pourraient être exposés à des modifications au runtime de trucs qui sont supposés être constants.

Reply:
Solved, income-tax est good en terme de variable par références retournés.